### PR TITLE
New article for Supabase integration

### PR DESCRIPTION
**Pull request by vcampitelli**
Creating a new article describing how to integrate with [Supabase](https://supabase.com).

PS: we do have some problems while rendering `JSX` content as `pygments` doesn't have support for it. There is a [jsx-lexer package](https://pypi.org/project/jsx-lexer/) that should handle it, though. I've used plain `javascript` tags for highlighting, which causes strange colors.

### CODE REVIEW

File: site/docs/v1/tech/oauth/integrations/supabase.adoc [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+After you finish, add `@supabase/supabase-js` as a dependency to the project:
+
+[source,shell]
+----
+$ npm install @supabase/supabase-js
+----
+
+== Issuing JWTs with Supabase's key
+
+Supabase needs [glossary]#JSON Web Tokens# that are signed with their own HMAC-SHA256 secret key. So, you'll have to issue a new token when calling their API to identify the underlying user. To do this, change `src/api/auth/[...nextauth].js` to add some link:https://next-auth.js.org/configuration/callbacks[callbacks], which are ways of personalizing your NextAuth.js session. We have provided some comments directly in the code below to better explain what is going on.
```
Comment (vcampitelli) [inline site/docs/v1/tech/oauth/integrations/supabase.adoc:104]: It actually is `src/pages/api/auth/[...nextauth].js`

File: site/docs/v1/tech/oauth/integrations/supabase.adoc [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+[source,ini]
+----
+NEXT_PUBLIC_SUPABASE_URL="<Project URL>"
+NEXT_PUBLIC_SUPABASE_ANON_KEY="<anon public key from Project API keys>"
+SUPABASE_SIGNING_SECRET="<JWT Secret from JWT Settings>"
+----
+
+== Testing
+
+It's finally time to wrap everything up and test the application! Start the web server by running `yarn dev` and browse to http://localhost:3000[localhost:3000]. Click [uielement]#Log in# to be redirected to the FusionAuth login page. Fill in the credentials from one of the users you have insert data to Supabase and you should be back to your application with a list of the events beloging to them. To check if everything is working accordingly, click [uielement]#Log out# and repeat the same process, but using another user and you should see a different set of events being listed.
```
Comment (vcampitelli) [inline site/docs/v1/tech/oauth/integrations/supabase.adoc:266]: Change to `npm run dev`

File: site/docs/v1/tech/oauth/integrations/supabase.adoc [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+description: A practical approach of using FusionAuth SSO capabilities in a Supabase application
+navcategory: login-methods
+---
+
+== Overview
+
+link:https://supabase.com/[Supabase] is an open source hosting service for Postgres databases. In this article, you will build a simple event management web application using FusionAuth, Supabase, and Next.js, a web application framework built on top of React.
+
+When a user tries to log in to your application, they will be redirected to the FusionAuth login page. After logging in, they will be back to your application, which will call the Supabase API to retrieve events for them. This request will have a JWT issued to that user, which Supabase will use to only return rows they have access to, using Postgres' link:https://www.postgresql.org/docs/current/ddl-rowsecurity.html[Row Level Security (RLS)] feature.
+
```
Comment (vcampitelli) [inline site/docs/v1/tech/oauth/integrations/supabase.adoc:13]: Missing "Configure FusionAuth" section

File: site/docs/v1/tech/oauth/integrations/supabase.adoc [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+== Fetching data
+
+You should fetch rows from your Supabase tables and pass them to the frontend page using Next's `getServerSideProps()` hook. Change `src/pages/index.js` and add the code below to check if the current session is valid and whether it has the issued JWT from before. If so, it'll call the Supabase API to retrieve data from the `events` table we created in the beginning of this article.
+
+[source,javascript]
+----
+import {createSupabaseClient} from '@/utils/supabase';
+import {authOptions} from '@/pages/api/auth/[...nextauth]';
+
```
Comment (vcampitelli) [inline site/docs/v1/tech/oauth/integrations/supabase.adoc:184]: Missing `getServerSession` import

File: site/docs/v1/tech/oauth/integrations/supabase.adoc [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+== Configuring environment variables
+
+As part of the process of setting up your Next.js application from our link:/blog/2023/02/10/nextjs-single-sign-on[Adding single sign-on to a Next.js app using OIDC] blog post, you should have already created a `.env.local` file. Now, it's time to paste the information you copied from Supabase API settings earlier into it, by adding the variables below and replacing everything that is quoted with the actual value.
+
+[source,ini]
+----
+NEXT_PUBLIC_SUPABASE_URL="<Project URL>"
+NEXT_PUBLIC_SUPABASE_ANON_KEY="<anon public key from Project API keys>"
+SUPABASE_SIGNING_SECRET="<JWT Secret from JWT Settings>"
+----
```
Comment (vcampitelli) [inline site/docs/v1/tech/oauth/integrations/supabase.adoc:262]: Missing `NEXTAUTH_SECRET`

**Review by mooreds — COMMENTED** [review]
Okay, I did a high level review, not a close one because there were a few high level things I wanted to address.

* I was really interested in writing doc that would get us listed here: https://supabase.com/docs/guides/auth/overview as a third party provider. This doesn't appear to be what this is. Am I missing something?
* The images all appear to be problematic. Is this dependent on the safari CSS changes you made here: https://github.com/FusionAuth/fusionauth-style/pull/25 ?
* For jsx, what would you recommend we do? Does kramdown (the highlighter that jekyll uses; pygments is used by asciidoc) support jsx?
* we don't typically reference blog posts in docs (other than in 'related posts' section), bc they expire. However, this whole doc feels like it would make a great blog post.
* please don't abbreviate anonymous to anon: "anon public key" should be "anonymous public key"
* for the JWT you have to re-issue, that's a bit weird. alternatives include importing the HMAC to FusionAuth and using a jwt populate lambda or using the jwt vend API with FusionAuth. Did you consider either of these?

All in all, I think this is a good explanation of how to integration FusionAuth with supabase, but I think I was looking for the auth integration to be a bit tighter, the way it is with, say, Auth0.

**Comment by vcampitelli** [comment]
- Docs: that was my first idea and I even started using their `.mdx` files, but after the "PagerDuty requirements" situation I thought we would prefer having it on our side to have more flexibility. But of course, it makes much more sense to have it there to spread the word. In the end, it seems that once again I tried to satisfy both scenarios and did something in between that didn't fit anything. [Their Auth0 docs](https://supabase.com/docs/guides/integrations/auth0) does mention an official blog post from their site though, so we could do something like that (having an article there and referencing our official and probably more up-to-date blog post/article). I can easily convert it to `.mdx` and make it closer to their structure.
- Images: It was this article that made me work on the Safari images, but I forgot to rebase the branch after finishing FusionAuth/fusionauth-style#25, so I'm doing that now.
- Syntax: yes, because we're using Rouge with kramdown and [it does support JSX](https://rouge-ruby.github.io/docs/file.Languages.html)
- Blog post: like I said, I can easily convert this to their `.mdx` files and structure and change this to become less "bloggy". It would only take me 2 hours. What do you think?
- Abbreviation: this is what Supabase have in their UI, exactly `anon public key`. I'll just put it inside a `[field]##` if we choose to maintain this as an article.
- JWT: yeah I also found it weird but their other articles do like that and I was carried away. It makes much more sense to import the key to our key master + the lambda function though, I think it didn't know about them back then. Let me rewrite this part to do those things instead.

**Comment by mooreds** [comment]
@vcampitelli sounds like a plan. Why don't you see what is involved in getting it into their docs.

Options I see:
* leave the content as is, move it to a blog post
* rewrite the content to be more doc-y, leave it in the FusionAuth docs
* add it to the supabase docs. Here's the clerk PR as an example: https://github.com/supabase/supabase/pull/6033

I'd prefer the last.

re: abbreviation, ah, then it's okay.

So, in short, let's make it like the Auth0 or clerk guide, and submit it to them as a PR. 

Make sense, @vcampitelli ?

**Comment by vcampitelli** [comment]
I've forked their repository and am rewriting the article in [FusionAuth/supabase](https://github.com/FusionAuth/supabase).

I'll be uploading PDFs to the [releases page](https://github.com/FusionAuth/supabase/releases) so you don't need to actually run the entire repository locally to see how the article is going to look.

**Comment by mooreds** [comment]
Sounds good, thanks for the update. Let me know when you need me to take a look.

**Comment by vcampitelli** [comment]
Closing this in favor of FusionAuth/supabase#1
